core/handler/convert_fields.go

```diff
@@ -176,3 +176,91 @@ func RunUnsafeCode(vm *otto.Otto, code string, timeOut time.Duration) (value ott
 	}()
 	return vm.Run(code)
 }
+
+type DownlinkFunctions struct {
+	// Encoder is a JavaScript function that accepts the payload as JSON and
+	// returns an array of bytes
+	Encoder string
+}
+
+func (f *DownlinkFunctions) Encode(payload map[string]interface{}) ([]byte, error) {
+	if f.Encoder == "" {
+		return nil, errors.New("Encoder function not set")
+	}
+
+	vm := otto.New()
+	vm.Set("payload", payload)
+	value, err := RunUnsafeCode(vm, fmt.Sprintf("(%s)(payload)", f.Encoder), timeOut)
+	if err != nil {
+		return nil, err
+	}
+
+	if !value.IsObject() {
+		return nil, errors.New("Encoder does not return an object")
+	}
+
+	if value.Class() != "Array" {
+		return nil, errors.New("Encoder should return an Array.")
+	}
+
+	v, err := value.Export()
+	if err != nil {
+		return nil, err
+	}
+
+	m, ok := v.([]int64)
+	if !ok {
+		return nil, errors.New("Encoder should return an Array of numbers")
+	}
+
+	res := make([]byte, len(m))
+	for i, v := range m {
+		if v < 0 || v > 255 {
+			return nil, errors.New("Encoder error: numbers in array should be between 0 and 255")
+		}
+		res[i] = byte(v)
+	}
+
+	return res, nil
+}
+
+func (f *DownlinkFunctions) ProcessDownlink(payload map[string]interface{}) ([]byte, bool, error) {
+	encoded, err := f.Encode(payload)
+	if err != nil {
+		return nil, false, err
+	}
+	// The handler as just to encode data without checking if it's valid or not.
+	return encoded, true, nil
+}
+
+func (h *handler) ConvertFieldsDown(ctx log.Interface, appDown *mqtt.DownlinkMessage, ttnDown *pb_broker.DownlinkMessage) error {
+
+	// The validity of the message is the application responsability.
+	if appDown.Fields == nil {
+		return nil
+	}
+
+	// Impossible to have fields and payload at the same time
+	if appDown.Payload != nil {
+		return errors.New("Error: both Fields and Payload provided")
```

Comment:
Remove `"Error: "`

Never a need to create an error saying that's an error, or give the function context because the caller knows